api/swagger.yml

```diff
@@ -468,6 +468,22 @@ components:
           type: boolean
           default: false
 
+    CommitOverrides:
+      type: object
+      properties:
+        message:
+          type: string
+          description: replace the commit message
+        metadata:
+          type: object
+          description: update the metadata of the commit record. New keys will be added to the original metadata, and existing keys will be overwritten.
```

Comment:
So it's not possible to remove a metadata key?
My suggestion is to avoid merging old and new metadata structs. If it exists here, it completely overrides the previous. Any fancy merging could be done in the client side and post the result here.